test/test_service.py

```diff
@@ -308,7 +308,7 @@ def find_available_port():
     raise RuntimeError('Cannot find an available port to listen on')
 
 
-def local_endpoint(port, endpoint, version='v1'):
+def local_endpoint(port, endpoint, version='v2'):
```

Comment:
Should we not have tests for both the v1 and the v2 endpoints, now that they both co-exist?

Reply:
Maybe.

I was thinking here that given that we "freeze" the v(n - 1) routes that theres no point carrying all that extra code to test them. But it would not be too hard to do that

Reply:
Let's please add them. I want to make sure the old routes do not regress when we change the newer ones, since the code is shared in the backend.

Reply:
@cosimoc Just a heads up that I am working on this, but its turning out to be a rather messy rebase given the other pushed changes. I'll have something ready later tomorrow.

Reply:
No problem, thanks for the heads up.